### PR TITLE
877 Supporting Arbitrary ORTB Params

### PR DESCRIPTION
**Pull request by jsligh**
Closes https://github.com/prebid/prebid-mobile-ios/issues/877 

Follows: https://docs.google.com/document/d/1cRWDNO6i7wLyQdQA9ZzmK0QumIr2vvC40r5GhHhOSz4/edit?usp=sharing 

Old PR: https://github.com/prebid/prebid-mobile-ios/pull/941

### CODE REVIEW

File: PrebidMobile/PrebidMobileRendering/ORTB/PBMORTBBidRequest.m [hunk 1]
```diff
@@ -64,15 +74,63 @@ - (nonnull PBMJsonDictionary *)toJsonDictionary {
...
+    
+    //merge with ortbConfig from SDK with priority away from SDK
+    ret = [self mergeDictionaries: ret joiningArgument2: ortbObj joiningArgument3: false];
     
     ret = [ret pbmCopyWithoutEmptyVals];
     
     return ret;
 }
 
+- (nonnull PBMMutableJsonDictionary *)mergeDictionaries:(NSMutableDictionary*)dictionary1 joiningArgument2:(NSMutableDictionary*)dictionary2
```
[the remark this answers is not in the document]
Comment (YuriyVelichkoPI) [reply PrebidMobile/PrebidMobileRendering/ORTB/PBMORTBBidRequest.m:107]: I understand it. I just can't predict the behavior and output of the algorithm depending on the different formats and structure of the input data.  I don't understand why we should not use the existing RTB objects and structure and instead rely on pure JSON operations. 

But if you think it is enough - no objections from me.

**Comment by bretg** [comment]
@YuriyVelichkoPI - you asked my review here -- was it just the issue about the possibility of specifying imps at the global level or something else?

**Comment by YuriyVelichkoPI** [comment]
Yes, I was asking about merge requirements. The current implementation relies on the correctness of the input and doesn't validate the RTB structure. In the current implementation the merge is based not on RTB but on JSON. Publishers can provide whatever they want. If we delegate all the final validation to the server in the current scope - ok, let's try.

**Review by bretg — APPROVED** [review]
My review indicates that I agree with the "low-validation" approach. Checking for valid JSON is a good idea. Checking for valid ORTB is not necessary.

I'm open to the idea of validating that `imp` is not specified in the global ORTB, but this is not necessary in my opinion.

**Pull request merged**